doc/virtual_batch_size_multi_gpu.md

```diff
@@ -0,0 +1,117 @@
+# Virual minibatch size and multiple GPUs
+
+Note: The code examples here should be interpreted as pseudo code, i.e. they should demonstrate what the code does. The actual implementation is more complicated, but a general understanding of the operating principle should be enough for most use cases.
+
+## What is (Virual) minibatch size?
+
+In `padertorch` you have two options to use a minibatch size for your model.
+
+The first possibility is to incude the batching in your train dataset (i.e. batch multiple examples together inside of your data preprocessing).
+In this case your model has to work on multiple examples and might have to deal with padding.
+The `padertorch.Trainer` will not recognice this kind of minibatch size, because the trainer simply forwards the examples from your dataset to the model:
+
+```python
+dataset = do_batching(dataset)                      # <-----
+for batch in dataset:                             # <-----
+    batch = model.example_to_device(batch)
+    review = model.review(batch, model(batch))  # <-----
+    loss = loss_from_review(review)
+    report_to_tensorboard(review)
+    loss.backward()
+    optimizer.step()
+    optimizer.zero_grad()
+```
+
+As an example how the batching can be done on the fly with `lazy_dataset` see `padertorch/contrib/examples/tasnet/train.py`.
+
+The second option is the `virtual_minibatch_size` argument of `padertorch.Trainer`.
+With this option you can increase the minibatch size without changing your dataset or your model. In this case, only the trainer is handling the batch size:
+
+```python
+i = 0                                               # <-----
+for example in dataset:
+    example = model.example_to_device(example)
+    review = model.review(example, model(example))
+    loss = loss_from_review(review)
+    report_to_tensorboard(review)
+    loss.backward()
+    i += 1
+    if i == virtual_minibatch_size:                 # <-----
+        i = 0                                       # <-----
+        optimizer.step()                            # <-----
+        optimizer.zero_grad()
+```
+
+Both options for the minibatch size can be combined.
+The effective minibatch size for the optimizer will be the dataset minibatch size times the `virtual_minibatch_size`. For operations like batch normalization (i.e. operations that work in the minibatch axis) the batch size will be the minibatch that is used to produce the dataset, NOT the virtual batch size.
+
+## Why use (Virtual) minibatch size?
```

Comment:
Mention that it is simliar to `accumulate_grads` (I think) from ESPnet?

Reply:
It's called `accum_grad` in ESPnet

Reply:
Du you want to add this the comparison with ESPnet?